Explorer/Assets/DCL/InWorldCamera/InWorldCamera/UI/InWorldCameraController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Arch.Core;
+using Cysharp.Threading.Tasks;
+using DCL.Audio;
+using DCL.CharacterCamera;
+using DCL.ExplorePanel;
+using DCL.InWorldCamera.CameraReelStorageService;
+using DCL.InWorldCamera.CameraReelStorageService.Schemas;
+using DCL.InWorldCamera.Playground;
+using DCL.UI;
+using ECS.Abstract;
+using JetBrains.Annotations;
+using MVC;
+using System.Diagnostics;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.InWorldCamera.UI
+{
+    /// <summary>
+    /// Handles Logic for the InWorldCamera HUD that appears when user enables InWorldCamera.
+    /// </summary>
+    public class InWorldCameraController : ControllerBase<InWorldCameraView>
+    {
+        private readonly Button sidebarButton;
+        private readonly World world;
+        private readonly IMVCManager mvcManager;
+        private readonly ICameraReelStorageService storageService;
+
+        private ScreencaptureShortcutsController shortcutsController;
+        private SingleInstanceEntity? cameraInternal;
+
+        private bool shortcutPanelIsOpen;
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Overlay;
+        private SingleInstanceEntity? camera => cameraInternal ??= world.CacheCamera();
+
+        public InWorldCameraController([NotNull] ViewFactoryMethod viewFactory, Button sidebarButton, World world, IMVCManager mvcManager, ICameraReelStorageService storageService) : base(viewFactory)
+        {
+            this.world = world;
+            this.mvcManager = mvcManager;
+            this.storageService = storageService;
+            this.sidebarButton = sidebarButton;
+
+            sidebarButton.onClick.AddListener(ToggleInWorldCamera);
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            viewInstance!.CloseButton.onClick.AddListener(RequestDisableInWorldCamera);
+            viewInstance.TakeScreenshotButton.onClick.AddListener(RequestTakeScreenshot);
+            viewInstance.CameraReelButton.onClick.AddListener(OpenCameraReelGallery);
+            viewInstance.ShortcutsInfoButton.onClick.AddListener(ToggleShortcutsInfo);
+
+            shortcutsController = new ScreencaptureShortcutsController(() => viewInstance.ShortcutsInfoPanel);
+            mvcManager.RegisterController(shortcutsController);
+        }
+
+        public override void Dispose()
+        {
+            viewInstance!.CloseButton.onClick.RemoveListener(RequestDisableInWorldCamera);
```

Comment:
a small doubt I have here as well but might be wrong, if for some reason we launch the application and during initial loading we stop the execution this Dispose will be called and might be called before the scene lifecycle started, meaning that viewInstance could be null.
To avoid dirty error messages you could use the conditional access ?